src/js/utils/xhr.ts

```diff
@@ -0,0 +1,51 @@
+import { RN_GLOBAL_OBJ } from './worldwide';
+
+/**
+ * The DONE ready state for XmlHttpRequest
+ *
+ * Defining it here as a constant b/c XMLHttpRequest.DONE is not always defined
+ * (e.g. during testing, it is `undefined`)
+ *
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState}
+ */
+export const XHR_READYSTATE_DONE = 4;
+
+let originalXhrOpen: XMLHttpRequest['open'] | null = null;
+let originalXhrSend: XMLHttpRequest['send'] | null = null;
+
+/**
+ * Saves original XHR methods so that they can be used later
+ */
+export function preserveXMLHttpRequest(customGlobal: { XMLHttpRequest?: typeof XMLHttpRequest } = RN_GLOBAL_OBJ): void {
+  if (!customGlobal.XMLHttpRequest || !customGlobal.XMLHttpRequest.prototype) {
```

Comment:
Another way to achieve this could also be like I did here for Spotlight: https://github.com/getsentry/spotlight/pull/335

that's fetch, but may also work here - basically we leverage the `__sentry_original__` prop we keep on monkeypatched APIs:

```ts
export function getNativeFetchImplementation(): FetchImpl {
  if (fetchIsWrapped(window.fetch)) {
    return window.fetch.__sentry_original__;
  }

  return window.fetch;
}

function fetchIsWrapped(fetchImpl: FetchImpl): fetchImpl is WrappedFetchImpl {
  return '__sentry_original__' in fetchImpl;
}
```

Reply:
Thank you, I'll use that.